src/_guides/openapi/augmenting-generated-openapi.md

```diff
@@ -0,0 +1,359 @@
+
+      - name: Deploy API documentation
+        uses: bump-sh/github-action@v1
+        with:
+          doc: 68ac0647-184a-4e9d-accc-682a5b1f7189
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.public.yaml
+```
+
+
+
+## MOdifying Servers with Overlays
+
+Before you send OpenAPI-based API documentation out to the public you may want to remove development and staging servers to avoid confusing end-users, and in the past that's been something that has caused conflict with dev teams. Do you force them to remove it from their OpenAPI, despite it being useful for them, or do you try and hack it later? 
+
+No more arguments, and no more hacks. You can do that with overlays.
+
+```yaml
+# overlays.yaml
+overlay: 1.0.0
+info:
+  title: Overlay to customise API for Protect Earth
+  version: 0.0.1
+actions:
+  - target: '$.servers.*'
+    description: Remove all other servers so we can add our own.
+    remove: true
+
+  - target: '$.servers'
+    description: Pop our server into the empty server array.
+    update:
+      - description: Production
+        url: https://api.protect.earth/
+```
+
+IF you'd like something more generic: 
+
+```yaml
+# overlays.yaml
+overlay: 1.0.0
+info:
+  title: Overlay to customise API for Protect Earth
+  version: 0.0.1
+actions:
+  - target: '$.servers[?(@.description=="Development" || @.description=="Staging")]'
+    description: Remove Development and Staging servers but leave anything else.
+    remove: true
+```
+
+This action will remove any servers with a `description` of "Development" or "Staging". This approach could also be done with openapi-filters if you pop `x-internal` on there, so there is a bit of crossover.
+
+## Summary
+
+Filters are a quick and relatively simple way to customize OpenAPI, but they can really only do the one thing. If you just want to remove some operations, tags, servers, etc., then openapi-filter is probably the way to go.
+
+Overlays are a lot more advanced and can do both removals and updates. It's a bit harder to work with, and the Speakeasy dependency is not the easiest to install on some computers (especially Continuous Integration environments), but the power is incredible.
+
+Being able to change things however you like and publish that off seamlessly is really handy, and will hopefully be the last time you need to do awkward JSON/YAML hacking on other peoples documents. JSONPath is a tricky thing to learn, but if you can master regex you can master JSONPath.
+
+
+
+## Further Thoughts - not for inclusion
+
+Beware how you split up files, because if your flag lives inside a `$ref` you will need to run the filter on that file directly, and that could lead to confusion. For example when I tried to use this structure I had a invalid $ref error... 
+```
+paths:
+  /upload:
+    $ref: paths/upload.yaml
+```
+
+You can get around this by [bundling first](https://docs.bump.sh/guides/openapi/advanced-ref-usage/), then filtering, then deploying.
```

Comment:
We try to push our users NOT to bundle `$ref`, because it creates huge openapi documents, is there no way to apply overlays on the root document which understands `$ref`? (and thus modifies the good targeted $ref file)

If not, we'll have to implement that somewhere are Bump maybe, most probably in the CLI

Reply:
I'm trying to suggest they bundle _in their toolchain_ but not keep that bundle around in git, so it would be one transitionary step before bump gets to it. 

I'm talking to @scharrier about trying to get the CLI supporting it, and the github action too, so this would all be a whoooooole lot easier, but even that would not necessarily solve the $ref issue as you'd have to convert those JSONPath's into "Bump In Memory Objects" somehow.

Reply:
> I'm talking to @scharrier about trying to get the CLI supporting overlays

You've been heard. Could you try this first draft of the new CLI command out: https://github.com/bump-sh/cli/pull/553 :pray:

Reply:
Very cool! Article updated to use the `bump overlay` command.

Reply:
Thanks a lot. I've added your article overlay example in the test suite of my branch to be able to validate all the use case describe here.

All of the examples work except from your target `$.info.contact` update to update on inexistent `contact:` object. I've had to modify the target to `$.info` and change the `update:` value to include the `contact:` key.

Reply:
Cool! I dont know if I was doing something wrong, or if its just a use case that the tool doesn't support in its current form, but I've updated my example to use something that works with the tool as it is.